lib/controllers/facebook-login.js

```diff
@@ -60,21 +60,27 @@ module.exports = function (req, res) {
 
       res.locals.user = expandedAccount;
       req.user = expandedAccount;
-      helpers.createIdSiteSession(req.user, req, res);
 
-      var nextUrl = url.parse(req.query.next || '').path || (resp.created ? config.web.register.nextUri : config.web.login.nextUri);
+      helpers.createStormpathSession(req.user, req, res, function (err) {
```

Comment:
I'd say it's out of scope for this PR. But Yeah, the Facebook and Google providers could probably be refactored to one. LinkedIn does some things slightly different, but that could probably also be refactored:)